Tournament.cpp

```diff
@@ -0,0 +1,263 @@
+#include "Tournament.h"
+#include <iostream>
+#include <fstream>
+#include <algorithm>
+
+Tournament::Tournament(int choice, std::string& path, std::string& pathGroup)
+{
+  const int EURO = 1;
+  const int WORLD = 2;
+  if (choice == EURO)
+  {
+	try
+	{
+	  addParticipants(path, m_euroParticipants);
+	  setGroup(pathGroup, m_numberEuroGroup);
+	}
+	catch (const std::exception& ex)
+	{
+	  std::cout << "Error! "<< ex.what() << std::endl;
+	}
+	
+  }
+  if (choice == WORLD)
+  {
+	try
+	{
+	  addParticipants(path, m_worldCupParticipants);
+	  setGroup(pathGroup, m_numberWorldCupGroup);
+	}
+	catch (const std::exception& ex)
+	{
+	  std::cout << "Error!" << ex.what() << std::endl;
+	}
+  }
+}
+
+void Tournament::addParticipants(std::string& path, int numberParticipants) //The function reads participants from a file.
+{
+  std::ifstream participantsList;
+  participantsList.open(path);
+  if (!participantsList.is_open() and !participantsList.good())
+	throw std::exception("Unable to add participants. File not found");
+  else
+  {
+	std::string name;
+	int rank = 0;
+	double fifa_points = 0;
+	for (int i = 0; i < numberParticipants; ++i)
+	{
+	  participantsList >> name >> rank >> fifa_points;
+	  auto newTeam = std::make_shared<Team>(name, rank, fifa_points);
+	  m_participants.push_back(newTeam);
+	}
+  }
+  participantsList.close();
```

Comment:
AFAIK ifstream destructor would automatically close the associated file, so since it's a local variable you can omit explicit closing. It doesn't hurt, though.